src/nvim/normal.c

```diff
@@ -2145,23 +2147,40 @@ void do_check_scrollbind(bool check)
   old_leftcol = curwin->w_leftcol;
 }
 
+static int get_scrollbind_amount(win_T *orig_wp, int old_topline, int topfill_diff)
+{
+  int orig_topline = orig_wp->w_topline;
+
+  if (orig_topline == old_topline) {
+    return -topfill_diff;
+  }
+
+  int count = -abs(topfill_diff);
+
+  int top = MIN(orig_topline, old_topline);
+  int bot = MAX(orig_topline, old_topline);
+  int buflen = orig_wp->w_buffer->b_ml.ml_line_count;
+  for (int i = top + 1; i <= bot; i++) {
+    count += plines_win(orig_wp, i, buflen);
```

Comment:
Using `plines_win()` may be incorrect here, as that counts the number of screen lines, while `scrollup()` and `scrolldown()` may scroll by screen lines or text lines depending on the value of the 'smoothscroll' option.

Actually, previously (and in Vim) the opposite problem existed, where the number of text lines is always used.

Reply:
I haven't put any work in to make this work for smooth scroll, which afaik is still a bit experimental/new.

Either way, using plines should still be better than using text lines as `scrollup()`/`scrolldown()` specifically handle topfill, so we need to send numbers to it which include filler lines.

What I'm not sure about is whether the number we pass should include folds or not.

Reply:
I was more talking about wrapped lines, not filler lines.

Reply:
I know, the changes here don't improve the situation for wrapped lines, but I don't think regresses it either. Or does it regress because plines respects wrapping? If so I can avoid using plines.

Reply:
I think it does regress. When a line is wrapped is one window, it's more common for the corresponding line to also be wrapped (and occupy the same number of screen lines) in the other window. When scrolling over such a line with 'nosmoothscroll', the old behavior works correctly, but the new behavior